Packs/RetarusSecureEmailGateway/Integrations/RetarusSecureEmailGateway/RetarusSecureEmailGateway.py

```diff
@@ -0,0 +1,286 @@
+        channel (str): Retarus channel to connect through.
+        token_id (str): Retarus token id.
+        fetch_interval (int): Time between fetch iterations.
+        verify_ssl (bool): Whether to verify ssl when connecting.
+
+    Yields:
+        EventConnection: eventConnection to receive events from.
+    """
+    extra_headers = {"Authorization": f"Bearer {token_id}"}
+
+    ssl_context = ssl.create_default_context()
+    if not verify_ssl:
+        ssl_context.check_hostname = False
+        ssl_context.verify_mode = ssl.CERT_NONE
+
+    try:
+        with connect("wss://" + url + f"/email/siem/v1/websocket?channel={channel}",
+                     additional_headers=extra_headers,
+                     ssl=ssl_context) as ws:
+            connection = EventConnection(
+                connection=ws,
+                fetch_interval=fetch_interval
+            )
+            set_the_integration_context(
+                "last_run_results", f"Opened a connection successfully at {datetime.now().astimezone(timezone.utc)}")
+            yield connection
+
+    except Exception as e:
+        set_the_integration_context("last_run_results",
+                                    f"{str(e)} \n This error happened at {datetime.now().astimezone(timezone.utc)}")
+        raise DemistoException(f"{str(e)}\n")
+
+
+def set_the_integration_context(key: str, val):  # pragma: no cover
+    """Adds a key-value pair to the integration context dictionary.
+        If the key already exists in the integration context, the function will overwrite the existing value with the new one.
+    """
+    cnx = demisto.getIntegrationContext()
+    cnx[key] = val
+    demisto.setIntegrationContext(cnx)
+
+
+def is_interval_passed(fetch_start_time: datetime, fetch_interval: int) -> bool:  # pragma: no cover
+    """Checks if the specified interval has passed since the given start time.
+        This function is used within the fetch_events function to determine if the time to fetch events is over or not.
+
+    Args:
+        fetch_start_time (datetime): The start time of the interval
+        fetch_interval (int): The interval in seconds
+
+    Returns:
+        bool: True if the interval has passed, False otherwise
+    """
+    return fetch_start_time + timedelta(seconds=fetch_interval) < datetime.now().astimezone(timezone.utc)
+
+
+def perform_long_running_loop(connection: EventConnection, fetch_interval: int):
+    """
+    Long running loop iteration function. Fetches events from the connection and sends them to XSIAM.
+
+    Args:
+        connection (EventConnection): A connection object to fetch events from.
+        fetch_interval (int): Fetch time for this fetching events cycle.
+    """
+    events = fetch_events(connection, fetch_interval)
+    demisto.debug(f'{LOG_PREFIX} Adding {len(events)} Events to XSIAM')
+
+    # Send the events to the XSIAM.
+    try:
+        send_events_to_xsiam(events, vendor=VENDOR, product=PRODUCT)
```

Comment:
Add debug log after sending to XSIAM in case of issues in the sending due to amount of data.